mqtt/mqtt-bridge/src/client.rs

```diff
@@ -410,16 +410,35 @@ impl ShutdownHandle {
     }
 }
 
+#[async_trait]
+pub trait ClientPublishHandle {
```

Comment:
we dont need this trait anymore, right?

Reply:
It is needed because of the mock! macro needed for clone to work.

Reply:
I checked docs and we could use something like this 
```rust
/// A client publish handle.
#[derive(Debug, Clone)]
pub struct PublishHandle(mqtt3::PublishHandle);

impl PublishHandle {
    pub async fn publish(&mut self, publication: Publication) -> Result<(), PublishError> {
        self.0.publish(publication).await
    }
}

mock! {
    pub PublishHandle {
        async fn publish(&mut self, publication: Publication) -> Result<(), PublishError>;
    }

    pub trait Clone {
        fn clone(&self) -> Self;
    }
}
```

Reply:
This won't work because the tests won't be able to resolve publish for the MockPublishHandle. From the docs:
https://docs.rs/mockall/0.8.3/mockall/macro.mock.html

> which basically involves repeating the struct's or trait's definitions.

Since we can't define impl in this mock! macro I was thinking that we needed a trait. I just tried it again to confirm and this won't work.

Reply:
I've just ran suggested above changes and tests passed.